### ISSUE TITLE
How to Create Blog Categories?

### ISSUE THREAD

**Issue by emmadigital** [netlify/netlify-cms#3685]
How should I create blog categories in Netlify CMS? Is the below code in config.yml file is the right way to create categories? How should I deal with Site URLs, do I need to define that in config.yml file:
Category
    - Category 1
    - Category 2
    - Category 3

 - name: "uncategorized"
    label: "Uncategorized"
    folder: "src/pages/blog/uncategorized"
    create: true
    slug: "{{year}}-{{month}}-{{day}}-{{slug}}"
    fields:
      - {label: "Template Key", name: "templateKey", widget: "hidden", default: "category"}
      - {label: "Title", name: "title", widget: "string"}
      - {label: "Publish Date", name: "date", widget: "datetime"}
      - {label: "Draft", name: "draft", widget: "boolean", default: true}
      - {label: "Categories", name: "categories", widget: "string", required: true}
      - {label: "Tags", name: "tags", widget: "list"}
      - {label: "Author", name: "author", widget: "hidden", default: "src/img/steve-smith-ardalis-200x200.jpg"}
      - {label: "Header Image", name: "image", widget: image, required: false}
      - {label: "Body", name: "body", widget: "markdown"}
      - {label: "Commenting", name: "comments", widget: boolean, default: true}
      - {label: "Type", name: "type", widget: hidden, default: post}
      - {label: "Sharing", name: "share", widget: boolean, default: true}   

I want to have below Url structure for blog category:

List of posts in Category Url: www.sitedomain.com/category/category-name
Single Blog Post Url: www.sitedomain.com/post-name

**Comment by buxzee** [comment]
I want to create below categories for my blog posts, how should I achieve this please advise.

Uncategorized
Productivity
Security
Personal

**Comment by emmadigital** [comment]
Error loading the CMS configuration
Config Errors:
YAMLSyntaxError: Document is not valid YAML (bad indentation?) at line 68, column 1:

- name: "categories"
^^^^^^^^^^^^^^^^^^^^…
Check your config.yml file.

**Comment by emmadigital** [comment]
Please advise what I am doing wrong? how should I create categories for my blog.

```
- name: "categories"
    label: "Categories"
    identifier_field: slug
    folder: "src/pages/blog/"
    create: true
    fields:
      - {label: "Category", name: "category_name", widget: "string"}
      - {label: "Category Image", name: "category_img", widget: "image"}
      - {label: "Slug", name: "slug", widget: "string"} 

    - name: "personal"
    label: "Personal"
    identifier_field: slug
    folder: "src/pages/blog/personal"
    create: true
    slug: "{{year}}-{{month}}-{{day}}-{{slug}}"
    fields:
      - {label: "Template Key", name: "templateKey", widget: "hidden", default: "category"}
      - {label: "Title", name: "title", widget: "string"}
      - {label: "Publish Date", name: "date", widget: "datetime"}
      - {label: "Draft", name: "draft", widget: "boolean", default: true}
      - {label: "Categories", name: "categories", widget: "string", required: true}
      - {label: "Tags", name: "tags", widget: "list"}
      - {label: "Author", name: "author", widget: "hidden", default: "src/img/steve-smith-ardalis-200x200.jpg"}
      - {label: "Header Image", name: "image", widget: image, required: false}
      - {label: "Body", name: "body", widget: "markdown"}
      - {label: "Commenting", name: "comments", widget: boolean, default: true}
      - {label: "Type", name: "type", widget: hidden, default: post}
      - {label: "Sharing", name: "share", widget: boolean, default: true}
```

**Comment by barthc** [comment]
@emmadigital  you file indentations are off. You can use this online tool to validate your file http://www.yamllint.com/.  Update your config file like so:
```yaml
collections:
  - name: "categories"
    label: "Categories"
    identifier_field: slug
    folder: "src/pages/blog/"
    create: true
    fields:
      - {label: "Category", name: "category_name", widget: "string"}
      - {label: "Category Image", name: "category_img", widget: "image"}
      - {label: "Slug", name: "slug", widget: "string"} 

  - name: "personal"
    label: "Personal"
    identifier_field: slug
    folder: "src/pages/blog/personal"
    create: true
    slug: "{{year}}-{{month}}-{{day}}-{{slug}}"
    fields:
      - {label: "Template Key", name: "templateKey", widget: "hidden", default: "category"}
      - {label: "Title", name: "title", widget: "string"}
      - {label: "Publish Date", name: "date", widget: "datetime"}
      - {label: "Draft", name: "draft", widget: "boolean", default: true}
      - {label: "Categories", name: "categories", widget: "string", required: true}
      - {label: "Tags", name: "tags", widget: "list"}
      - {label: "Author", name: "author", widget: "hidden", default: "src/img/steve-smith-ardalis-200x200.jpg"}
      - {label: "Header Image", name: "image", widget: image, required: false}
      - {label: "Body", name: "body", widget: "markdown"}
      - {label: "Commenting", name: "comments", widget: boolean, default: true}
      - {label: "Type", name: "type", widget: hidden, default: post}
      - {label: "Sharing", name: "share", widget: boolean, default: true}
```

**Comment by barthc** [comment]
> I want to have below Url structure for blog category:
List of posts in Category Url: www.sitedomain.com/category/category-name
Single Blog Post Url: www.sitedomain.com/post-name

Not sure I understand the OP of the issue. The cms cannot control the url structure of the site. If the we talking about implementing a tag feature, then the relation widget or a simple list widget.

**Comment by emmadigital** [comment]
I checked my config.yml file on http://www.yamllint.com/ I get this error:https://prnt.sc/sals5v whats wrong with my code: Please advise

```
backend:
  name: git-gateway
  branch: master
  commit_messages:
    create: 'Create {{collection}} “{{slug}}”'
    update: 'Update {{collection}} “{{slug}}”'
    delete: 'Delete {{collection}} “{{slug}}”'
    uploadMedia: '[skip ci] Upload “{{path}}”'
    deleteMedia: '[skip ci] Delete “{{path}}”'

publish_mode: editorial_workflow
media_folder: static/img
public_folder: /img

collections:
  - name: "blog"
    label: "Blog"
    folder: "src/pages/blog"
    create: true
    slug: "{{year}}-{{month}}-{{day}}-{{slug}}"
    fields:
      - {label: "Template Key", name: "templateKey", widget: "hidden", default: "blog-post"}
      - {label: "Title", name: "title", widget: "string"}
      - {label: "Publish Date", name: "date", widget: "datetime"}
      - {label: "Description", name: "description", widget: "text"}
      - {label: "Featured Post", name: "featuredpost", widget: "boolean"}
      - {label: "Featured Image", name: "featuredimage", widget: image}
      - {label: "Body", name: "body", widget: "markdown"}
      - {label: "Tags", name: "tags", widget: "list"}

  - name: "pages"
    label: "Pages"
    files:
      - file: "src/pages/index.md"
        label: "Landing Page"
        name: "index"
        fields:
          - {label: "Template Key", name: "templateKey", widget: "hidden", default: "index-page"}
          - {label: Title, name: title, widget: string}
          - {label: Image, name: image, widget: image}
          - {label: Heading, name: heading, widget: string}
          - {label: Subheading, name: subheading, widget: string}
          - {label: Mainpitch, name: mainpitch, widget: object, fields: [{label: Title, name: title, widget: string}, {label: Description, name: description, widget: text}]}
          - {label: Description, name: description, widget: string}
          - {label: Intro, name: intro, widget: object, fields: [{label: Heading, name: heading, widget: string}, {label: Description, name: description, widget: text}, {label: Blurbs, name: blurbs, widget: list, fields: [{label: Image, name: image, widget: image}, {label: Text, name: text, widget: text}]}]}
          - {label: Main, name: main, widget: object, fields: [{label: Heading, name: heading, widget: string}, {label: Description, name: description, widget: text}, {label: Image1, name: image1, widget: object, fields: [{label: Image, name: image, widget: image}, {label: Alt, name: alt, widget: string}]}, {label: Image2, name: image2, widget: object, fields: [{label: Image, name: image, widget: image}, {label: Alt, name: alt, widget: string}]}, {label: Image3, name: image3, widget: object, fields: [{label: Image, name: image, widget: image}, {label: Alt, name: alt, widget: string}]}]}
      - file: "src/pages/about/index.md"
        label: "About"
        name: "about"
        fields:
          - {label: "Template Key", name: "templateKey", widget: "hidden", default: "about-page"}
          - {label: "Title", name: "title", widget: "string"}
          - {label: "Body", name: "body", widget: "markdown"}
      - file: "src/pages/products/index.md"
        label: "Products Page"
        name: "products"
        fields:
          - {label: "Template Key", name: "templateKey", widget: "hidden", default: "product-page"}
          - {label: Title, name: title, widget: string}
          - {label: Image, name: image, widget: image}
          - {label: Heading, name: heading, widget: string}
          - {label: Description, name: description, widget: string}
          - {label: Intro, name: intro, widget: object, fields: [{label: Heading, name: heading, widget: string}, {label: Description, name: description, widget: text}, {label: Blurbs, name: blurbs, widget: list, fields: [{label: Image, name: image, widget: image}, {label: Text, name: text, widget: text}]}]}
          - {label: Main, name: main, widget: object, fields: [{label: Heading, name: heading, widget: string}, {label: Description, name: description, widget: text}, {label: Image1, name: image1, widget: object, fields: [{label: Image, name: image, widget: image}, {label: Alt, name: alt, widget: string}]}, {label: Image2, name: image2, widget: object, fields: [{label: Image, name: image, widget: image}, {label: Alt, name: alt, widget: string}]}, {label: Image3, name: image3, widget: object, fields: [{label: Image, name: image, widget: image}, {label: Alt, name: alt, widget: string}]}]}
          - {label: Testimonials, name: testimonials, widget: list, fields: [{label: Quote, name: quote, widget: string}, {label: Author, name: author, widget: string}]}
          - {label: Full_image, name: full_image, widget: image}
          - {label: Pricing, name: pricing, widget: object, fields: [{label: Heading, name: heading, widget: string}, {label: Description, name: description, widget: string}, {label: Plans, name: plans, widget: list, fields: [{label: Plan, name: plan, widget: string}, {label: Price, name: price, widget: string}, {label: Description, name: description, widget: string}, {label: Items, name: items, widget: list}]}]}
 - name: "iraq"
    label: "Iraq"
    identifier_field: slug
    folder: "src/pages/blog/iraq"
    create: true
    slug: "{{year}}-{{month}}-{{day}}-{{slug}}"
    fields:
      - {label: "Template Key", name: "templateKey", widget: "hidden", default: "category"}
      - {label: "Title", name: "title", widget: "string"}
      - {label: "Publish Date", name: "date", widget: "datetime"}
      - {label: "Draft", name: "draft", widget: "boolean", default: true}
      - {label: "Categories", name: "categories", widget: "string", required: true}
      - {label: "Tags", name: "tags", widget: "list"}
      - {label: "Author", name: "author", widget: "hidden", default: "src/img/steve-smith-ardalis-200x200.jpg"}
      - {label: "Header Image", name: "image", widget: image, required: false}
      - {label: "Body", name: "body", widget: "markdown"}
      - {label: "Commenting", name: "comments", widget: boolean, default: true}
      - {label: "Type", name: "type", widget: hidden, default: post}
      - {label: "Sharing", name: "share", widget: boolean, default: true}
  
```

**Comment by emmadigital** [comment]
I have created categories in Content Manager But its not getting deployed. Below is the build log:

12:13:16 AM: Waiting for other deploys from your team to complete
12:13:37 AM: Build ready to start
12:13:39 AM: build-image version: 2dbd444fcdce00cf06325060a8238d5ae3e86774
12:13:39 AM: build-image tag: v3.3.7
12:13:39 AM: buildbot version: d305548225ac57bb00eb5a596b0b8674e76a4681
12:13:39 AM: Fetching cached dependencies
12:13:39 AM: Starting to download cache of 263.4MB
12:13:42 AM: Finished downloading cache in 3.323371146s
12:13:42 AM: Starting to extract cache
12:13:57 AM: Finished extracting cache in 14.385248251s
12:13:57 AM: Finished fetching cache in 17.773995782s
12:13:57 AM: Starting to prepare the repo for build
12:13:57 AM: Preparing Git Reference refs/heads/master
12:13:58 AM: Found Netlify configuration file netlify.toml in site root
12:13:58 AM: Found Netlify configuration file(s). Overriding site configuration
12:13:58 AM: Different functions path detected, going to use the one specified in the Netlify configuration file: 'lambda' versus '' in the Netlify UI
12:13:58 AM: Creating functions prep folder
12:13:58 AM: Starting build script
12:13:58 AM: Installing dependencies
12:13:59 AM: Started restoring cached node version
12:14:03 AM: Finished restoring cached node version
12:14:04 AM: v12.16.3 is already installed.
12:14:04 AM: Now using node v12.16.3 (npm v6.14.4)
12:14:04 AM: Attempting ruby version 2.6.2, read from environment
12:14:06 AM: Using ruby version 2.6.2
12:14:06 AM: Using PHP version 5.6
12:14:06 AM: Started restoring cached node modules
12:14:06 AM: Finished restoring cached node modules
12:14:06 AM: Started restoring cached go cache
12:14:06 AM: Finished restoring cached go cache
12:14:06 AM: unset GOOS;
12:14:06 AM: unset GOARCH;
12:14:06 AM: export GOROOT='/opt/buildhome/.gimme/versions/go1.12.linux.amd64';
12:14:06 AM: export PATH="/opt/buildhome/.gimme/versions/go1.12.linux.amd64/bin:${PATH}";
12:14:06 AM: go version >&2;
12:14:06 AM: export GIMME_ENV='/opt/buildhome/.gimme/env/go1.12.linux.amd64.env';
12:14:06 AM: go version go1.12 linux/amd64
12:14:06 AM: Installing missing commands
12:14:06 AM: Verify run directory
12:14:06 AM: Executing user command: npm run build
12:14:07 AM: > gatsby-starter-netlify-cms@1.1.3 build /opt/build/repo
12:14:07 AM: > npm run clean && gatsby build
12:14:07 AM: > gatsby-starter-netlify-cms@1.1.3 clean /opt/build/repo
12:14:07 AM: > gatsby clean
12:14:07 AM: info Deleting .cache, public
12:14:07 AM: info Successfully deleted directories
12:14:10 AM: success open and validate gatsby-configs - 0.040s
12:14:11 AM: success load plugins - 1.475s
12:14:11 AM: success onPreInit - 0.014s
12:14:11 AM: success delete html and css files from previous builds - 0.017s
12:14:11 AM: success initialize cache - 0.011s
12:14:11 AM: success copy gatsby files - 0.042s
12:14:11 AM: success onPreBootstrap - 0.010s
12:14:11 AM: success createSchemaCustomization - 0.032s
12:14:11 AM: success source and transform nodes - 0.343s
12:14:12 AM: success building schema - 0.527s
**12:14:12 AM: error Your site's "gatsby-node.js" created a page with a component that doesn't exist.
12:14:12 AM: The path to the missing component is "/opt/build/repo/src/templates/null.js"
12:14:12 AM: The page object passed to createPage:
12:14:12 AM: {
12:14:12 AM:     "path": "/blog/iraq/",
12:14:12 AM:     "tags": null,
12:14:12 AM:     "component": "/opt/build/repo/src/templates/null.js",
12:14:12 AM:     "context": {
12:14:12 AM:         "id": "ea971b2b-07b0-5bf4-b7a6-97576b2a3a0b"
12:14:12 AM:     }
12:14:12 AM: }**
12:14:12 AM: See the documentation for the "createPage" action — https://www.gatsbyjs.org/docs/actions/#createPage
12:14:12 AM: not finished createPages - 0.047s
12:14:12 AM: npm
12:14:12 AM:  ERR! code ELIFECYCLE
12:14:12 AM: npm ERR! errno 1
12:14:12 AM: npm ERR! gatsby-starter-netlify-cms@1.1.3 build: `npm run clean && gatsby build`
12:14:12 AM: npm ERR! Exit status 1
12:14:12 AM: npm ERR!
12:14:12 AM: npm ERR! Failed at the gatsby-starter-netlify-cms@1.1.3 build script.
12:14:12 AM: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
12:14:12 AM: npm
12:14:12 AM:  ERR! A complete log of this run can be found in:
12:14:12 AM: npm ERR!     /opt/buildhome/.npm/_logs/2020-05-04T21_14_12_570Z-debug.log
12:14:12 AM: Function Dir: /opt/build/repo/lambda
12:14:12 AM: TempDir: /tmp/zisi-5eb0856b5384ba0006f9cbd2
12:14:13 AM: Prepping functions with zip-it-and-ship-it 0.3.1
12:14:13 AM: [ { path: '/tmp/zisi-5eb0856b5384ba0006f9cbd2/hello.zip',
12:14:13 AM:     runtime: 'js' } ]
12:14:13 AM: Prepping functions complete
12:14:13 AM: Caching artifacts
12:14:13 AM: Started saving node modules
12:14:13 AM: Finished saving node modules
12:14:13 AM: Started saving pip cache
12:14:13 AM: Finished saving pip cache
12:14:13 AM: Started saving emacs cask dependencies
12:14:13 AM: Finished saving emacs cask dependencies
12:14:13 AM: Started saving maven dependencies
12:14:13 AM: Finished saving maven dependencies
12:14:13 AM: Started saving boot dependencies
12:14:13 AM: Finished saving boot dependencies
12:14:13 AM: Started saving go dependencies
12:14:13 AM: Finished saving go dependencies
12:14:13 AM: Error running command: Build script returned non-zero exit code: 1
12:14:13 AM: Failing build: Failed to build site
12:14:14 AM: failed during stage 'building site': Build script returned non-zero exit code: 1
12:14:14 AM: Finished processing build request in 35.303780801s

My gatsby-node.js file https://github.com/emmadigital/ardalis-newsite/blob/master/gatsby-node.js
